openff/toolkit/topology/topology.py

```diff
@@ -1175,9 +1175,13 @@ def _identify_chemically_identical_molecules(
                 if mol2_idx in already_matched_mols:
                     continue
                 mol2 = self.molecule(mol2_idx)
-                are_isomorphic, atom_map = Molecule.are_isomorphic(
-                    mol1, mol2, return_atom_map=True
-                )
+                if isinstance(mol1, type(mol2)) and isinstance(mol2, type(mol1)):
+                    are_isomorphic, atom_map = mol1.__class__.are_isomorphic(
```

Comment:
(not blocking) I think we can call staticmethods from any instance of the class, so this may work and remove the unsightly `.__class__` bit

```suggestion
                    are_isomorphic, atom_map = mol1.are_isomorphic(
```